app/scripts/views/permissions.js

```diff
@@ -0,0 +1,141 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+define([
+  'cocktail',
+  'underscore',
+  'views/form',
+  'views/base',
+  'stache!templates/permissions',
+  'lib/promise',
+  'lib/auth-errors',
+  'lib/constants',
+  'lib/url',
+  'views/mixins/resend-mixin',
+  'views/mixins/back-mixin',
+  'views/mixins/service-mixin'
+],
+function (Cocktail, _, FormView, BaseView, Template, p, AuthErrors, Constants,
+    URL, ResendMixin, BackMixin, ServiceMixin) {
+  //var t = BaseView.t;
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'confirm',
+
+    initialize: function (options) {
+      // Account data is passed in from sign up and sign in flows.
+      var data = this.ephemeralData();
+      this._account = data && this.user.initAccount(data);
+
+      this.type = options.type;
+    },
+
+    getAccount: function () {
+      return this._account;
+    },
+
+    context: function () {
+      var account = this.getAccount();
+      var serviceUri = URL.getOrigin(this.relier.get('redirectUri'))
+                        .replace(/https?:\/\//, '');
+
+      return {
+        email: account.get('email'),
+        serviceName: this.relier.get('serviceName'),
+        termsUri: this.relier.get('termsUri'),
+        privacyUri: this.relier.get('privacyUri'),
+        serviceUri: serviceUri
+      };
+    },
+
+    events: {
+      // validateAndSubmit is used to prevent multiple concurrent submissions.
+      'click #proceed': BaseView.preventDefaultThen('validateAndSubmit')
+    },
+
+    beforeRender: function () {
+      // user cannot proceed if they have not initiated a sign up/in.
+      if (! this.getAccount().get('sessionToken')) {
+        this.navigate(this._previousScreen());
+        return false;
+      }
+    },
+
+    submit: function () {
+      var self = this;
+      var account = self.getAccount();
+
+      self.logScreenEvent('proceed');
+
+      if (self.is('sign_up')) {
+        return self.onSignUpSuccess(account);
+      } else if (account.get('verified')) {
+        return self.onSignInSuccess(account);
+      } else {
+        return self.onSignInUnverified(account);
+      }
+    },
+
+    onSignUpSuccess: function (account) {
```

Comment:
Perhaps to avoid duplication with the signin/signup modules, these three functions could be extracted into one ore two mixins (one for sign in, one for sign up? overkill?) that are mixed in where needed?